juju/charmstore
# Use the external debugstatus library.

Also refactor the debug/status endpoint to avoid
repeating similar checks.

Update the dependencies accordingly.

Move the status tests to their own status_test.go file,
and implemented the assertStatusCall test helper.

QA:
- start mongo;
- make server;
- visit http://localhost:8080/v4/debug/status and
  ensure the stats make sense.
